src/main/java/com/networknt/schema/JsonSchema.java

```diff
@@ -82,7 +82,7 @@ private JsonSchema(ValidationContext validationContext, String schemaPath, URI c
 
             if (config.isOpenAPI3StyleDiscriminators()) {
                 ObjectNode discriminator = (ObjectNode) schemaNode.get("discriminator");
-                if (null != discriminator) {
+                if (null != discriminator && null != validationContext.getCurrentDiscriminatorContext()) {
```

Comment:
Needed for the case where the schema holds a discriminator, but it is not referenced from an `anyOf`, so is actually disabled. Why would you want to support this?

Say you have `Person` extended `Husband` and `Wife` with `Person` having a `spouse` parameter and a `discriminator`. Then you might want to use the `discriminator` only in a nested occurrence